component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -545,6 +545,22 @@ private void validateUsingSubject(String subject, AuthenticationContext authenti
             addOrValidateCertificate(subject, authenticationContext, data, claims, cert);
         }
     }
+    /**
+     * check and validate the tenant domain of the user.
+     *
+     * @param userName               user name of the authenticated user.
+     */
+    private String getAuthenticUserName(String userName){
+
+        String authenticUser = null;
+        String tempUser = null;
+        if (userName.contains("carbon.super")) {
```

Comment:
Why only carbon.super has been checked here?

Reply:
The tenant's user names come along with the tenant domain (test@tenant1.com/test@wso2.com@gmailcom) and super tenant users come as the test@carbon.super. As per this fix when we config a cert for a particular user we need to give the cert CN value as the user name and if the user is in super tenant the CN value can be configured without the tenant domain (ex- CN: test) and for other tenants, it needs to config with the tenant domain (CN- test@tenant1.com). So that for comparison with the user name and cert's CN value we need to remove the super tenant part from the user name value since it retrieves from the "authenticatedUser.getAuthenticatedSubjectIdentifier()" method. then the CN value and username value which can be compared according to the authentication steps.

Reply:
we can pass authenticated user to the method getAuthenticatedUserName, so that if the username has carbon.super then we can get the username from authenticated.getusername and assign it rather than splittting.